src/mocompiler.js

```diff
@@ -16,37 +16,42 @@ export default function (table) {
 }
 
 /**
- * Creates a MO compiler object.
- *
- * @constructor
- * @param {import('./index.d.ts').GetTextTranslations} table Translation table as defined in the README
- * @return {import('./index.d.ts').Compiler} Compiler
+ * Prepare the header object to be compatible with MO compiler
+ * @param {{[key: string]: string}} headers the headers
+ * @return {{[key: string]: string}} The prepared header
  */
-function Compiler (table = {}) {
-  this._table = table;
-
-  let { headers = {}, translations = {} } = this._table;
-
-  headers = Object.keys(headers).reduce((result, key) => {
+function prepareMoHeaders (headers) {
+  return Object.keys(headers).reduce((/** @type {{[key: string]: string}} */ result, key) => {
     const lowerKey = key.toLowerCase();
 
     if (HEADERS.has(lowerKey)) {
       // POT-Creation-Date is removed in MO (see https://savannah.gnu.org/bugs/?49654)
       if (lowerKey !== 'pot-creation-date') {
-        result[HEADERS.get(lowerKey)] = headers[key];
+        const value = HEADERS.get(lowerKey);
+        if (value) {
+          result[value] = headers[key];
+        }
       }
     } else {
       result[key] = headers[key];
     }
 
     return result;
   }, {});
+}
 
-  // filter out empty translations
-  translations = Object.keys(translations).reduce((result, msgctxt) => {
+/**
+ * Prepare the translation object to be compatible with MO compiler
+ * @param {import('./types.js').GetTextTranslations['translations']} translations
+ * @return {import('./types.js').GetTextTranslations['translations']}
```

Comment:
If you use the suggestion above, I would also declare the `Translations` type like this:

```js
// At the top of the file with the other type import. We can use `GetTextTranslations` directly because it's already imported.
/**
 * @param {GetTextTranslations['translations']} Translations
 */
```

```suggestion
 * @param {Translations} translations
 * @return {Translations}
```

I haven't checked, though if `Translations` could be typed separately, rather than looking it up with key access on `GetTextTranslations`, that would be preferred. If that is acceptable then it would change to:

```js
// At the top of the file with the other type import. We can use `GetTextTranslations` directly because it's already imported.
/**
 * @typedef {import('./types.js').Translations) Translations
 */
```

Reply:
yes be patient but that is the fastest way with my IDE to generate jsdocs types. a 'cleaner' commit will follow all these changes

Reply:
i just give a try and seems that jsDocs only supports inline import :/ 

![image](https://github.com/smhg/gettext-parser/assets/8550908/bcaee1ef-fe80-4568-a2f6-742901236ff3)

Reply:
It looks to me like you didn't import them. Is it okay for me to PR this branch so you can see what I mean?

Reply:
sure! I think you have even permission to edit my repo, if not tell me